### PR TITLE
fix: allow package version to be non-semver

### PR DESCRIPTION
**Pull request by yaronschwimmer**
- [X] Ready for review
- [X] Follows CONTRIBUTING rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?
When comparing metadata files versions, we want to avoid crashing when a version is not a semver valid version. We now coerce the version so we'll be able to compare it safely.

### CODE REVIEW

File: lib/python-parser/metadata-parser.ts [hunk 1]
```diff
@@ -29,7 +30,12 @@ export function getPackageInfo(fileContent: string): PythonPackage {
       }
     }
   }
-  return { name: name.toLowerCase(), version, dependencies } as PythonPackage;
+  const validVersion = semver.coerce(version);
+  return {
+    name: name.toLowerCase(),
+    version: validVersion ? validVersion.toString() : version,
```
Comment (tommyknows) [inline lib/python-parser/metadata-parser.ts:36]: Does it make sense to fallback to the original version if the semver library can't parse it? 
Or should we potentially throw an error instead? What's the failure mode here? :)

Comment (tommyknows) [reply lib/python-parser/metadata-parser.ts:36]: That sounds to me like you should throw a (custom?) error and handle that case in the calling code to discard this file? 😄

Comment (tommyknows) [reply lib/python-parser/metadata-parser.ts:36]: I guess that's the JS way :P

Comment (yaronschwimmer) [reply lib/python-parser/metadata-parser.ts:36]: Both ways work :)

Comment (yaronschwimmer) [reply lib/python-parser/metadata-parser.ts:41]: I went with your suggestion

File: lib/python-parser/metadata-parser.ts [hunk 2]
```diff
@@ -29,7 +30,17 @@ export function getPackageInfo(fileContent: string): PythonPackage {
...
-  return { name: name.toLowerCase(), version, dependencies } as PythonPackage;
+  const validVersion = semver.coerce(version);
+  if (!validVersion) {
+    throw new PythonInvalidVersionError(
+      `version ${version} is not compatible with semver and cannot be compared`,
+    );
+  }
+  return {
+    name: name.toLowerCase(),
+    version: validVersion ? validVersion.toString() : version,
```
Comment (tommyknows) [inline lib/python-parser/metadata-parser.ts:41]: I guess we don't need that check then, if we throw an error :)

File: test/unit/python-metadata-parser.spec.ts [hunk 3]
```diff
@@ -72,4 +71,55 @@ describe("python metadata parser", () => {
...
+      Requires-Dist: colorama ; platform_system == "Windows"
+      Requires-Dist: importlib-metadata ; python_version < "3.8"
+    `;
+    const packageResult = getPackageInfo(fileContent);
+    expect(packageResult.name).toEqual("click");
+    expect(packageResult.version).toEqual("8.3.0");
+    expect(packageResult.dependencies).toHaveLength(2);
+  });
+
+  it("falis to parse package metadata when version can't be coerced to semver", () => {
```
Comment (tommyknows) [inline test/unit/python-metadata-parser.spec.ts:101]: NiT: `falis` typo :)

File: lib/python-parser/metadata-parser.ts (hunk 2 above)
Comment (yaronschwimmer) [reply lib/python-parser/metadata-parser.ts:41]: oh, you're right. forgot to delete

**Review by tommyknows — APPROVED** [review]

**Pull request merged**